src/main/kotlin/com/exactpro/th2/check1/rule/AbstractCheckTask.kt

```diff
@@ -152,7 +171,9 @@ abstract class AbstractCheckTask(
                     } else {
                         legacy.executorService
                     }
-                checkTask.begin(legacy.lastSequence, executor)
+                legacy.sequenceData.apply {
+                    checkTask.begin(this.lastSequence, this.lastMessageTimestamp, executor, this.untrusted)
```

Comment:
You don't need to use `this` here. Because of `apply` the correct context should be chosen